core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StorageFolder.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.container.storage;
+
+import java.nio.file.Path;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public interface StorageFolder {
```

Comment:
It's not only to wrap the path, because in the `ServerStorage` implementation of this interface, there is a lock mechanism to prevent concurrent reads and writes.
I created 2 implementations to have a version with locking and a version without, to use respectively when working with subfolders of the storage or when updating the whole storage